src/shared/steps/setup-device-step/deviceInfo.js

```diff
@@ -0,0 +1,44 @@
+import { isMobile, isMobileOnly, mobileVendor } from 'react-device-detect';
+
+let deviceInfo;
+
+const getDeviceType = async () => {
+    if (!navigator.getBattery) {
+        return 'desktop';
+    }
+
+    const battery = await navigator.getBattery();
+
+    if (battery.charging && battery.chargingTime === 0) {
+        return 'desktop';
+    }
+
+    return 'laptop';
+};
+
+const getDeviceInfo = async () => {
+    // isMobile returns true if device type is mobile or tablet
+    if (isMobile) {
+        const type = isMobileOnly ? 'mobile' : 'tablet';
+        const mobileName = mobileVendor !== 'none' ? mobileVendor : type;
+
+        return {
+            type,
+            nameSufix: mobileName,
+        };
+    }
+
+    const type = await getDeviceType();
+
+    deviceInfo = {
+        type,
+        nameSufix: type,
+    };
+
+    return deviceInfo;
+};
+
+getDeviceInfo();
```

Comment:
Can the promise fail?

Reply:
I've changed `getDeviceType()` itself to not fail. This way, `getDeviceInfo()` will never fail.